extension/source/Controllers/Keyring/KeyringController.ts

```diff
@@ -33,6 +34,33 @@ export default class KeyringController
     return this.state.wallets.map((x) => x.address);
   }
 
+  setHDPhrase(phrase: string) {
+    this.update({
+      HDPhrase: phrase,
+    });
+  }
+
+  isOnboardingComplete = (): boolean => {
+    return this.state.HDPhrase !== '';
+  };
+
+  async createHDAccount(): Promise<string> {
+    const mnemonic = this.state.HDPhrase;
+
+    if (mnemonic === '') {
+      const { phrase } = ethers.Wallet.createRandom().mnemonic;
+      this.setHDPhrase(phrase);
+    }
+
+    const node = ethers.utils.HDNode.fromMnemonic(mnemonic);
```

Comment:
See comment above - if `mnemonic` was an empty string before it'll still be an empty string here.